src/NewsService.Business/FindNewsCommand.cs

```diff
@@ -16,38 +34,186 @@ public class FindNewsCommand : IFindNewsCommand
+
+      return null;
+    }
 
     public FindNewsCommand(
       INewsRepository repository,
       INewsInfoMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      IDepartmentInfoMapper departmentInfoMapper,
+      ILogger<GetNewsCommand> logger,
+      IBaseFindRequestValidator baseFindValidator)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _rcGetImages = rcGetImages;
+      _logger = logger;
+      _baseFindValidator = baseFindValidator;
+      _departmentInfoMapper = departmentInfoMapper;
     }
 
-    public FindResultResponse<NewsInfo> Execute(FindNewsFilter findNewsFilter)
+    public async Task<FindResultResponse<NewsInfo>> Execute(FindNewsFilter findNewsFilter)
     {
       FindResultResponse<NewsInfo> response = new();
 
-      response.Body = _repository
-        .Find(findNewsFilter, out int totalCount)
-        .Select(_mapper.Map)
-        .ToList();
-
-      response.TotalCount = totalCount;
-
-      if (response.Errors.Any())
+      if (!_baseFindValidator.ValidateCustom(findNewsFilter, out List<string> errors))
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
         response.Status = OperationResultStatusType.Failed;
+        response.Errors = errors;
         return response;
       }
 
+      List<DbNews> dbNewsList = _repository.Find(findNewsFilter, out int totalCount);
+      if (dbNewsList == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+
+        response.Errors = new() { "News was not found." };
+        response.Status = OperationResultStatusType.Failed;
+      }
+
+      List<DepartmentData> departments = await GetDepartments(
+        dbNewsList.Where(d => d.DepartmentId.HasValue).Select(d => d.DepartmentId.Value).Distinct().ToList(),
+        response.Errors);
+      List<DepartmentInfo> departmentsInfo = departments.Select(_departmentInfoMapper.Map).ToList();
+
+      List<UserData> authors = await GetAuthors(
+        dbNewsList.Select(a => a.AuthorId).Distinct().ToList(),
+        response.Errors);
+
+      List<Guid> imagesIds = new();
+      imagesIds.AddRange(authors?.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList());
+      List<ImageData> avatarImages = await GetImages(imagesIds, response.Errors);
+
+      response.Body = dbNewsList
+        .Select(dbNews => _mapper.Map(
+          dbNews,
+          departmentsInfo.FirstOrDefault(d => dbNews.DepartmentId == d.Id),
+          authors.FirstOrDefault(a => dbNews.AuthorId == a.Id),
+          avatarImages))
```

Comment:
add user avatar in user map, not here, it is news mapper

Reply:
done